tests/test_orquestra_device.py

```diff
@@ -84,6 +100,112 @@ def test_empty_apply(self):
 
         assert dev.apply([]) is None
 
+    @pytest.mark.parametrize("keep", [True, False])
+    def test_keep_workflow_file(self, keep, tmpdir, monkeypatch):
+        """Test the option for keeping/deleting the workflow file."""
+
+        file_name = "test_workflow.yaml"
+        dev = qml.device("orquestra.forest", wires=3, keep_files=keep)
+        mock_res_dict = {"First": {"expval": {"list": [123456789]}}}
+        test_uuid = "1234"
+
+        assert not os.path.exists(tmpdir.join(f"expval-{test_uuid}.yaml"))
+        assert not dev.filenames
+        with monkeypatch.context() as m:
+            m.setattr(pennylane_orquestra.cli_actions, "user_data_dir", lambda *args: tmpdir)
+
+            # Disable submitting to Orquestra by mocking Popen
+            m.setattr(subprocess, "Popen", lambda *args, **kwargs: MockPopen())
+            m.setattr(
+                pennylane_orquestra.orquestra_device,
+                "loop_until_finished",
+                lambda *args, **kwargs: mock_res_dict,
+            )
+
+            # Disable random uuid generation
+            m.setattr(uuid, "uuid4", lambda *args: test_uuid)
+
+            @qml.qnode(dev)
+            def circuit():
+                qml.PauliX(0)
+                return qml.expval(qml.PauliZ(0))
+
+            assert circuit() == 123456789
+            file_kept = os.path.exists(tmpdir.join(f"expval-{test_uuid}.yaml"))
+            assert file_kept if keep else not file_kept
+            assert dev.filenames == ([f"expval-{test_uuid}.yaml"] if keep else [])
+            assert dev.latest_id == "SomeWorkflowID"
```

Comment:
Where does the "SomeWorkflowID" name come from? :thinking:

Reply:
Was in `conftest`, though pretty hidden. Created `test_workflow_id_name` and now that's imported here.